docs/site/migration/boot-scripts.md

```diff
@@ -6,8 +6,278 @@ sidebar: lb4_sidebar
+  constructor(
+    // inject `app` if you need access to other artifacts by `await this.app.get()`
+    @inject(CoreBindings.APPLICATION_INSTANCE) private app: Application,
+    // inject a datasource with key `datasources.${dsName}`
+    @inject('datasources.db') private ds: juggler.DataSource,
+  ) {}
+
+  /**
+   * This method will be invoked when the application starts
+   */
+  async start(): Promise<void> {
+    // Add your logic for start
+    console.log('This is a migrated synchronous boot script.');
+    console.log(`Your app has a datasource called: ${this.ds.name}`);
+  }
+
+  /**
+   * This method will be invoked when the application stops
+   */
+  async stop(): Promise<void> {
+    // Add your logic for stop
+    console.log('print message observer has stopped.');
+  }
+}
+```
+
+For the other script `server/boot/create-sample-promise.js`, the creation of its
+corresponding observer is exactly the same as `server/boot/print-info.js`. To
+avoid duplicate content, the steps are omitted here. The script retrieves
+LoopBack 3 model `Todo` and creates a sample through `Todo.create()`. While in
+LoopBack 4, model only describes the shape of data, and repository processes the
+persistence-related behavior. Therefore you need to call
+`TodoRepository.create(sample)` instead to create the sample. To migrate the
+`Todo` model, you can follow the steps in
+[Migrating model definitions and built-in APIs](./models/core.md).
+
+The migrated boot script as `CreateSampleObserver`'s start function:
+
+_The code snippet is the complete observer file, the LB3 boot script logic is
+only in the `start` function_
+
+```ts
+import {
+  /* inject, Application, CoreBindings, */
+  lifeCycleObserver, // The decorator
+  LifeCycleObserver,
+  inject,
+  CoreBindings,
+  Application, // The interface
+} from '@loopback/core';
+import {TodoRepository} from '../repositories';
+
+/**
+ * This class will be bound to the application as a `LifeCycleObserver` during
+ * `boot`
+ */
+@lifeCycleObserver()
+export class CreateSampleObserver implements LifeCycleObserver {
+  constructor(
+    // inject `app` if you need access to other artifacts by `await this.app.get()`
+    @inject(CoreBindings.APPLICATION_INSTANCE) private app: Application,
+    // inject a repository with key `repositories.${repoName}`
+    // or with the shortcut injector:
+    // `@repository(TodoRepository) private todoRepo: TodoRepository`
+    @inject('repositories.TodoRepository') private todoRepo: TodoRepository,
+  ) {}
+
+  /**
+   * This method will be invoked when the application starts
+   */
+  async start(): Promise<void> {
+    // Add your logic for start
+    console.log('This is a migrated asynchronous boot script');
+
+    const sample = {title: 'a todo sample', desc: 'Something to do.'};
+    // Create the sample by calling TodoRepo.create()
+    const result = await TodoRepo.create(sample);
+    console.log(`Sample created as: ${result}`);
```

Comment:
AFAICT from the CI logs, this line prints the following output:

```
Sample created as: [object Object]
```

`[object Object]` is not very useful. I am proposing to rewrite this line as follows:

```suggestion
    console.log('Sample created as', result);
```